src/intervals/interval_operations/boolean.jl

```diff
@@ -0,0 +1,228 @@
+"""
+function isstrictless(a::F, b::F) where {F<:Interval}
+    isempty(a) && isempty(b) && return true
+    (isempty(a) || isempty(b)) && return false
+    return isweaklylessprime(a.lo, b.lo) && isweaklylessprime(a.hi, b.hi)
+end
+
+"""
+    strictprecedes(a, b)
+
+Checks if the interval `a` is strictly to the left of interval `b`.
+
+Implement the `strictPrecedes` function of the IEEE Std 1788-2015 (Table 10.3).
+"""
+function strictprecedes(a::F, b::F) where {F<:Interval}
+    (isempty(a) || isempty(b)) && return true
+    return a.hi < b.lo
+end
+
+"""
+    isdisjoint(a,b)
+
+Checks if all the points of the interval `a` are within the interior of
+interval `b`.
+
+Implement the `disjoint` function of the IEEE Std 1788-2015 (Table 9.3).
+"""
+function isdisjoint(a::Interval, b::Interval)
+    (isempty(a) || isempty(b)) && return true
+    return isweaklylessprime(b.hi, a.lo) || isweaklylessprime(a.hi, b.lo)
+end
+
+function isdisjoint(a::Complex{F}, b::Complex{F}) where {F<:Interval}
+    return isdisjoint(real(a),real(b)) || isdisjoint(imag(a),imag(b))
+end
+
+"""
+    in(x, a)
+    ∈(x, a)
+
+Checks if the number `x` is a member of the interval `a`, treated as a set.
+
+Implement the `isMember` function of the IEEE Std 1788-2015 (section 10.6.3).
+"""
+function in(x::Real, a::Interval)
+    isinf(x) && return false
+    return a.lo <= x <= a.hi
+end
+
+in(x::Interval, y::Interval) = throw(ArgumentError("$x ∈ $y is not defined, maybe you meant `⊂`"))
+in(x::Real, a::Complex{F}) where {F<:Interval} = x ∈ real(a) && 0 ∈ imag(a)
+in(x::Complex{T}, a::Complex{F}) where {T<:Real, F<:Interval} = real(x) ∈ real(a) && imag(x) ∈ imag(a)
+
+contains_zero(x::Interval{T}) where T = zero(T) ∈ x
+
+isempty(x::Interval) = (inf(x) == Inf && sup(x) == -Inf)
+isentire(x::Interval) = (inf(x) == -Inf && sup(x) == Inf)
+isbounded(x::Interval) = (isfinite(x.lo) && isfinite(x.hi)) || isempty(x)
+isunbounded(x::Interval) = !isbounded(x)
+
+"""
+    isthin(x)
+
+Checks if `x` is the set consisting of a single exactly
+representable float. Any float which is not exactly representable
+does *not* yield a thin interval. Corresponds to `isSingleton` of
+the standard.
+"""
+isthin(x::Interval) = (x.lo == x.hi)
+
+"""
+    iscommon(x)
+
+Checks if `x` is a **common interval**, i.e. a non-empty,
+bounded, real interval.
+"""
+function iscommon(x::Interval)
+    (isentire(x) || isempty(x) || isunbounded(x)) && return false
+    return true
+end
+
+"""
+    isatomic(x::Interval)
+
+Check whether an interval `x` is *atomic*, i.e. is unable to be split.
+This occurs when the interval is empty, or when the upper bound equals the lower
+bound or the bounds are consecutive floating point numbers.
+"""
+isatomic(x::Interval) = isempty(x) || (x.lo == x.hi) || (x.hi == nextfloat(x.lo))
+
+"""
+    isthinzero(x)
+
+Return whether the interval only contains zero.
+"""
+isthinzero(x::Interval) = iszero(x.lo) && iszero(x.hi)
```

Comment:
Should this function be/correspond to `iszero`? I'm aware that `iszero` is defined such that it returns a three-way logic, though it seems to me that that is not necessary...

Reply:
It is used in many places internally e.g. to take the shortcut `x^(0..0) == 1`. So I think it is good to keep it.